### PR TITLE
BO-187 Upgrade Java to 1.8

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

File: modules/java/manifests/jre.pp [hunk 1]
```diff
@@ -1,9 +1,25 @@
-class java::jre {
+class java::jre (
+  $headless = false
+) {
 
-  require 'apt'
+  if $::facts['lsbdistid'] == 'Debian' {
+    $apt_sources = 'apt::source::backports'
```
Comment (tomaszdurka) [inline modules/java/manifests/jre.pp]: Let's directly include here for code navigation, etc.

File: modules/java/manifests/jre.pp [hunk 2]
```diff
@@ -1,9 +1,25 @@
-class java::jre {
+class java::jre (
+  $headless = false
+) {
 
-  require 'apt'
+  if $::facts['lsbdistid'] == 'Debian' {
```
Comment (tomaszdurka) [inline modules/java/manifests/jre.pp]: Let's use switch/case and throw in default case?

**Comment by tomaszdurka** [comment]
Tests are failing.

**Comment by ppp0** [comment]
retest this please

**Pull request merged**